clang/test/SemaSYCL/device_global.cpp

```diff
@@ -33,6 +33,65 @@ namespace {
 device_global<int> same_name; // OK
 }
 
+// TODO: add diagnostic for this case
+inline namespace other {
+device_global<int> same_name; // ILLEGAL: shadows "device_global" variable
+} // namespace other
+
+// TODO: add diagnostic for this case
+// inline namespace {
+//  namespace foo {               // ILLEGAL: namespace name shadows "::foo"
+//  }                             // namespace which contains "device_global"
+//                                // variable.
+//}
+
+struct BBar {
+private:
+  struct BarInsider {
+    static device_global<float> c;
+  };
+};
+
+struct ABar {
+  void method() {
+    // expected-error@+1{{'device_global' variables must be static or declared at namespace scope}}
```

Comment:
I think this error can be made more user-friendly. Since this is technically a static variable. User might not understand why this is an error since error says - I think the error - '....is not publicly accessible from namespace scope' suffices for all cases? WDYT?

Reply:
Also, should probably change `is not` to `should be`.